### PR TITLE
Before review

### PR DESCRIPTION
**Pull request by sameerdutta9703**
Please review Task1 and Task2

### CODE REVIEW

File: DEP Tasks/Trainers Task/JSCore/Task 2/Partition On.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// partition the items array so that all values for which pred returns true are
+// at the end, returning the index of the first true value
+function partitionOn(pred, items) {
+    var partIndex=0;
+    for(var i=0;i<items.length;i++)
```
Comment (itaphuviinnay) [inline DEP Tasks/Trainers Task/JSCore/Task 2/Partition On.js]: This solutions works fine but try to do it with build-in array methods .

File: DEP Tasks/Trainers Task/JSCore/Task 2/Word Count.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    // str = str.replace(/[ ]{2,}/gi," ");
+    // str = str.replace(/\n /,"\n");
+    // return str.split(" ").length;
+    
+    //.trim() - this removes all the spaces before and after the words
+    //.split() -  split it into an array upon the whitespace characters
+    //\s - this is for the white space characters
+    //filter - this takes a function and checks true and false and filters accordingly
+    //Boolean - this is a function that checks all are true or not, here it is used to remove undefined values
+     return str.trim().split(/\s/).filter(Boolean).length;
```
Comment (itaphuviinnay) [inline DEP Tasks/Trainers Task/JSCore/Task 2/Word Count.js]: Remove the commented code.

File: DEP Tasks/Trainers Task/JSCore/Task 1/Transportation on vacation.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+function rentalCarCost(d) {
+    // Your solution here
+    if(d>=7){
+      return amount=(d*40)-50;
```
Comment (itaphuviinnay) [inline DEP Tasks/Trainers Task/JSCore/Task 1/Transportation on vacation.js]: Where did `amount` variable has been declared?

Should use single return statement in a function where ever possible. So, remove multiple return statements.

File: DEP Tasks/Trainers Task/JSCore/Task 1/Basic Mathematical Operations.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+function basicOp(operation, value1, value2)
+{
+  // Code
+  if(operation==='+')
```
Comment (itaphuviinnay) [inline DEP Tasks/Trainers Task/JSCore/Task 1/Basic Mathematical Operations.js]: For every conditions there should be '{ }' braces.

**Review by itaphuviinnay — CHANGES_REQUESTED** [review]
Good work. Follow the below tip for the current and future pull requests

- Make sure to format the files before you commit. If you are using VSCode as your editor, use `ALT+SHIFT+F` for formatting the files. This makes the indentation of code proper and adds a new line at the end of the files which is a good practice

File: DEP Tasks/Trainers Task/JSCore/Task 1/Transportation on vacation.js (hunk 3 above)
Comment (sameerdutta9703) [reply DEP Tasks/Trainers Task/JSCore/Task 1/Transportation on vacation.js]: done

File: DEP Tasks/Trainers Task/JSCore/Task 1/Basic Mathematical Operations.js (hunk 4 above)
Comment (sameerdutta9703) [reply DEP Tasks/Trainers Task/JSCore/Task 1/Basic Mathematical Operations.js]: done

File: DEP Tasks/Trainers Task/JSCore/Task 2/Word Count.js (hunk 2 above)
Comment (sameerdutta9703) [reply DEP Tasks/Trainers Task/JSCore/Task 2/Word Count.js]: done

**Pull request merged**